### DISCUSSION TITLE
Deprecation warning in Hugo 0.112.0 potentially means breaking changes in DoIt

### DISCUSSION THREAD

**Discussion by CXwudi** [HEIGE-PCloud/DoIt#975] [General]
The story started from https://github.com/gohugoio/hugo/issues/10620

Before 0.112.0, if you have a group of configs, let's take the `[params.app]` as the example: 

https://github.com/HEIGE-PCloud/DoIt/blob/35323e4e9b97aa0793bb5b906195aa2a7314a686/exampleSite/config/_default/params.en.toml#L7-L24

And let's say we only want the `title` to be different in Chinese and English.

What we need to do previously is to copy everything into the `params.en.toml` and `params.zh-cn.toml`, otherwise it won't work.

Now, starting from 0.112.0, you can just copy the `title` into the corresponding `params.en.toml` and `params.zh-cn.toml` and leave the rest of the config in `params.toml`. The new merging behaviour will sort this out correctly.

Thanks to this behaviour, setting up i18n config is significantly simplified.

But the behaviour changes also deprecate some config structure. If you run `hugo server` on the `exampleSite`, it will show the warning⚠:

```
WARN 2023/05/28 00:40:16 config: languages.en.googleanalytics: custom params on the language top level is deprecated and will be removed in a future release. Put the value below [languages.en.params]. See https://gohugo.io/content-management/multilingual/#changes-in-hugo-01120

Re-run Hugo with the flag --panicOnWarning to get a better error message.
WARN 2023/05/28 00:40:16 config: languages.zh-cn.googleanalytics: custom params on the language top level is deprecated and will be removed in a future release. Put the value below [languages.zh-cn.params]. See https://gohugo.io/content-management/multilingual/#changes-in-hugo-01120

Re-run Hugo with the flag --panicOnWarning to get a better error message.
``` 

Basically, https://gohugo.io/content-management/multilingual/#changes-in-hugo-01120 said that we should no longer read a parameter from a specific language code (like `languages.en.googleanalytics` in this example), this deprecation also happens if you have `author.toml` to be i18n. 

Now the real problem is, if we fix all these deprecations, do we have to force users to use Hugo 0.112.0? Should we fix these deprecations or not

**Comment by HEIGE-PCloud** [comment]
Thanks for your post! After some careful inspection, no breaking change is needed on DoIt side. The only option that's causing problem is the `googleanalytics` option, which is deprecated and not used by DoIt long ago. 

I have updated the config structure for the exampleSite, and I expect current users can upgrade their Hugo version safely without much trouble.

**Reply by CXwudi** [reply]
Just a reminder that if `author.toml` is in i18n (`author.en.toml` and `author.zh-cn.toml`), the same deprecation happens.

But I honestly I am not sure is `author.toml` still used? Given that we already have `data/authors/**` for multi-author supports.

**Reply by HEIGE-PCloud** [reply]
The warning I get is

```
languages.en.name: custom params on the language top level is deprecated and will be removed in a future release. Put the value below [languages.en.params]. See https://gohugo.io/content-management/multilingual/#changes-in-hugo-01120
```

which isn't true, the configs for authors should live under `.Site.Author.*`. I think I need to look further...

**Reply by HEIGE-PCloud** [reply]
I think this is actually a Hugo issue. If I don't use `author.toml`, but just put the `[author]` field under `hugo.toml` instead, 

```toml
# hugo.toml
[author]
name = "PCloud"
# ...
```

```toml
# hugo.en.toml
[author]
name = "PCloud-EN"
# ...
```

```toml
# hugo.zh-cn.toml
[author]
name = "PCloud-CN"
# ...
```

Everything works as expected without warning.

**Reply by CXwudi** [reply]
Interesting 🤔, might worth to test it again with newest Hugo and rise an issue. I will try it when I am free
